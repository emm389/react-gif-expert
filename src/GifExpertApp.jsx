import { useState } from 'react';
import { AddCategory, GifGrid } from './components';

export const GifExpertApp = () => {

    //TODO: WE DON'T PUT OUR HOOKS INSIDE CONDITIONS LIKE IF, NEVER!!!

    const [categories, setCategories] = useState(['']);

    const onAddCategory = (newCategory) => {
        if (categories.includes(newCategory)) return;
        setCategories([newCategory, ...categories]);
        // setCategories([...categories, 'Halo']);
        // setCategories(cat => [...cat, 'Halo']);
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory 
                // categories={categories} setCategories={setCategories}
                onNewCategory={onAddCategory}
            />
            {
                categories.map(category => (
                    <GifGrid 
                        key={category}
                        category={category}
                    />
                ))
            }
        </>
    )
}

export default GifExpertApp
