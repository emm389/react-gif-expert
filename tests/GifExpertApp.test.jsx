import { render, screen } from "@testing-library/react";
import GifExpertApp from "../src/GifExpertApp";

describe('Tests on GifExpertApp', () => {
    test('should ', () => {
        const {container} = render(<GifExpertApp/>);
        screen.debug();

        expect(container).toMatchSnapshot();
    }); 
});
