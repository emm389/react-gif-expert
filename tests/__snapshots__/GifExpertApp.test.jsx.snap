// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on GifExpertApp should  1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search Gifs"
      type="text"
      value=""
    />
  </form>
  <h3 />
  <h2>
    Loading...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
